### ISSUE TITLE
Add the same external storage S3 bucket configuration twice 

### ISSUE THREAD

**Issue by ohlr** [nextcloud/server#18663]
### Steps to reproduce
1. Add the same external storage configuration (S3) twice (with different folder and different user)

### Expected behaviour
Do not allow that

### Actual behaviour
Cpu utilization goes to 100%
external storage does not allow put requests anymore
server restart necessary

### Server configuration
snap 

**Operating system:**
Ubuntu 18

**Web server:**
EC2 1CPU 1GB Ram 512MB for NC

**Comment by kesselb** [comment]
> Add the same external storage configuration (S3) twice (with different folder and different user)

Could you rephrase? Is different user a other nextcloud account or a another user with access to the s3 bucket?

**Comment by ohlr** [comment]
I created another user (next cloud account) and assigned it a bucket which was used by another user already.

I guess it would also crash if you only have one user and mount the same external bucket twice with different folder names.

**Comment by kesselb** [comment]
![image](https://user-images.githubusercontent.com/3902676/72359381-77e78780-36ee-11ea-8c39-2dc77173e007.png)

A quick test with my development setup seems to work. But the snap is version 16.0.5 and development is 18.0.0. Probably something fixed in the meantime.

**Comment by ohlr** [comment]
Okay, did you put data into the folders? And did you also try that with two different nextcloud accounts?

The problem is that NC let's you create these configurations but afterwards can't handle them.

Maybe the version is a problem indeed. I moved to S3 as primary storage, so it does not affect me anymore. Feel free to close if you can't reproduce.

**Comment by kesselb** [comment]
> Okay, did you put data into the folders? 

Yes.

>  And did you also try that with two different nextcloud accounts?

No.

> The problem is that NC let's you create these configurations but afterwards can't handle them.

Could you be more specific? What is broken? ;)

**Comment by ohlr** [comment]
The server cpu went up to 100%, and I couldn't put objects. But I did not elaborate deeper.

**Comment by ohlr** [comment]
What I wonder, if both folders have the same underlying s3 bucket, what is the desired behavior. Do they sync each other, what happens if files collide? 

I.e. in your case what happens if you create two files: Test1/test.txt Test2/test.txt.

**Comment by kesselb** [comment]
> what is the desired behavior.

I guess the desired behavior is not specified. It depends on the caching flag somehow. Without caching external storages should be in sync. With caching probably everything is possible. But a create operation will always check if the file is already present.

**Comment by ohlr** [comment]
The thing is now, if that works with multiple users they can overwrite data from other users.

**Comment by ohlr** [comment]
I don't see any scenario in which such a configuration is desired. I.e. it should be disabled.

**Comment by kesselb** [comment]
> I don't see any scenario in which such a configuration is desired. I.e. it should be disabled.

Group A: Read + Write access to bucket
Group B: Read access to bucket

You can configure this with the `files_accesscontrol` app or mount the bucket 2x and make on of the read only. Probably there are even more scenarios. But a note somewhere that mounting the same bucket more than once and set Check for changes to Never is probably bad.

**Comment by ohlr** [comment]
> Group A: Read + Write access to bucket
> Group B: Read access to bucket
Good point
